Documentation/Input/CreateDataProvider.md

```diff
@@ -0,0 +1,202 @@
+# Creating an input system data provider
+
+The Mixed Reality Toolkit input system is an extensible system for enabling input device support. To add support for a new hardware platform,
+a custom input data provider may be required.
+
+This article describes how to create custom data providers, also called device managers, for the input system. The example code shown here is
+from the [`WindowsMixedRealityDeviceManager`](xref:Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityDeviceManager).
+
+> The complete code used in this example can be found in the MixedRealityToolkit.Providers\WindowsMixedReality folder.
+
+## Namespace and folder structure
+
+> [!Important]
+> If an input system data provider is being submitted to the [Mixed Reality Toolkit repository](https://github.com/Microsoft/MixedRealityToolkit-Unity), the
+namespace **must** begin with Microsoft.MixedReality.Toolkit (ex: Microsoft.MixedReality.Toolkit.WindowsMixedReality) and the code should be
+located in a folder beneath MixedRealityToolkit.Providers (ex: MixedRealityToolkit.Providers\WindowsMixedReality).
+
+### Namespace
+
+Data providers are required to have a namespace to mitigate potential name collisions. It is recommended that the namespace includes the following components.
+
+- Company name
+- Feature area
+
+For example, an input data provider created by the Contoso company may be "Contoso.MixedReality.Input".
+
+### Recommended folder structure
+
+It is recommended that the source code for data providers be layed out in a folder heirarchy as shown in the following image.
+
+![Example folder structure](../Images/SpatialAwareness/ExampleProviderFolderStructure.png)
+
+Where ContosoInput contains the implementation of the data provider, the Editor folder contains the inspector (and any other Unity editor specific code) and Profiles
```

Comment:
The Screenshot does not show ContosoInput

Reply:
had the wrong image link

Reply:
updated images in latest commit